content/blog/2024-02-18-announcing-poetry-1-8-0.md

```diff
@@ -0,0 +1,110 @@
+
+### Better support for non-package projects
+
+Poetry is a tool for packaging and dependency management. Yet, it is perfectly fine
+if you want to use Poetry only for dependency management and not for packaging. However,
+you had to jump through some hoops to make this work. With Poetry 1.8, you can just
+specify that your project is not supposed to be packaged by setting the `package-mode`
+to `false`:
+
+```bash
+[tool.poetry]
+package-mode = false
+```
+
+With this setting:
+
+- all the [metadata](/docs/pyproject) that is required for packaging is not mandatory anymore
+- Poetry will not try to install the root package (similar to `poetry install --no-root`)
+- building and publishing will not be possible
+
+### Fetching metadata faster via PEP 658 support and new `lazy-wheel` feature
+
+When resolving dependencies with a cold cache, most of the time is spent for fetching
+metadata. In the past, Poetry had to download wheels to extract metadata (except for
+PyPI if metadata was available via its custom JSON API). Poetry 1.8 now supports
+[PEP 658](https://peps.python.org/pep-0658/) to fetch metadata without having to
+download wheels.
+
+However, most package indices do not support PEP 658 yet. To mitigate this,
+Poetry 1.8 introduces a new `lazy-wheel` feature, which is enabled by default. With
+this feature, Poetry will extract the metadata from remote wheels via HTTP range requests.
+This way, Poetry can fetch metadata by only downloading a small amount of the wheel
+instead of the whole file.
+
+{{% note %}}
+If the package index does not even provide hashes, Poetry still
+has to download the whole file to calculate the hash itself.
+{{% /note %}}
+
+Although we tested this feature with various package indices,
+there still might be servers that respond to HTTP range requests in an unexpected way.
+Therefore, if you experience any issues with this feature,
+you can [disable lazy-wheel](docs/master/configuration#solverlazy-wheel).
+
+## Other important Changes
+
+### Deprecating `default` source type
+
+Since Poetry 1.5, a warning is printed that the implicit PyPI source will be disabled
+automatically in the future if at least one custom primary source is configured.
+With Poetry 1.8, this warning becomes true. As a result, the `default` source type
+is not required anymore because a default source is exactly the same as the first primary
+source. This is the last deprecation in a series of changes to simplify and improve
+Poetry's handling of package sources.
+
+### Upgrading warning about inconsistent lockfile to an error
+
+In the past, Poetry only printed a warning when running `poetry install` and the lockfile
+was inconsistent with the pyproject.toml. This happened if dependencies were changed
+in the pyproject.toml without running `poetry lock --no-update`.
+As a result, `poetry install` might fail or succeed with expected or unexpected results.
+This warning has been upgraded to an error to avoid issues that are hard to debug.
```

Comment:
Is there a better way to explain this change? I read it multiple times, but I still have no idea about what the change is.

Reply:
There is certainly always room for improvement but I don't know how to explain it better. A warning becomes an error. In other words, the command will fail with return code unequal zero. If you have never seen the mentioned warning, it is probably irrelevant to you.

Reply:
I believe everyone using poetry saw the warning multiple times 🙂 I'm not sure why, but I think that "upgrade" comes with other changes, and the text was missing it. After rereading it, it feels more clear now. Let me share my other confusions, and the jury can decide if they need any updates:
* I'm missing how `--no-update` parameter is relevant. I checked the PR and the discussions but couldn't find any reference. The only thing I saw was `"Run `poetry lock [--no-update]` to fix it.` on an unedited line, which I believe makes better sense, emphasizing that it's just an optional parameter.
* `This happened if dependencies were changed in the pyproject.toml without running` This only covers a small portion of the relevant cases shared by people on the PR. My understanding is the issue and the solution is more about the hash from lock file not matching the relevant fields of the `pyproject.toml` file, not which file was edited by whom or for which purpose.

Reply:
I rephrased the paragraph also considering the changed wording from python-poetry/poetry#8962. I hope it is clearer now. I still believe that changing something from warning to error can be called an upgrade but I am open to better wording. 😄